packages/wonder-blocks-modal/src/components/modal-launcher.js

```diff
@@ -158,11 +173,33 @@ export default class ModalLauncher extends React.Component<Props, State> {
 
     handleCloseModal: () => void = () => {
         this.setState({opened: false}, () => {
-            this.props.onClose && this.props.onClose();
+            const {onClose, closedFocusId, schedule} = this.props;
+            const lastElement = this.lastElementFocusedOutsideModal;
+
+            onClose && onClose();
 
-            if (this.lastElementFocusedOutsideModal != null) {
-                // return focus to the element that triggered the modal
-                this.lastElementFocusedOutsideModal.focus();
+            // Focus on the specified element after closing the modal.
+            if (closedFocusId) {
+                const focusElement = (ReactDOM.findDOMNode(
+                    document.getElementById(closedFocusId),
+                ): any);
+
+                if (focusElement) {
+                    // Wait for the modal to leave the DOM before trying
+                    // to focus on the specified element.
+                    schedule.animationFrame(() => {
+                        focusElement.focus();
+                    });
+                    return;
+                }
+            }
+
+            if (lastElement != null) {
+                // Wait for the modal to leave the DOM before trying to
+                // return focus to the element that triggered the modal.
+                schedule.animationFrame(() => {
+                    lastElement.focus();
+                });
```

Comment:
Nit: Taking another look at this code, I can see that you could extract this returning focus part into a separate function (the `onClose` part could remain here). Some benefits about that:

- It reduces indentation and improves readability.
- Abstracts the "return focus" logic into a separate logic that can help for future refactors.

wdyt?